core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataManager.java

```diff
@@ -100,6 +108,17 @@ public void initialise() throws InitialisationException {
     return exceptionHandledMetadataFetch(componentId, MetadataAware::getMetadataKeys, EXCEPTION_RESOLVING_METADATA_KEYS);
   }
 
+  @Override
+  public MetadataResult<Map<String, Set<MetadataKey>>> getMetadataKeys(ConfigurationId configurationId) {
```

Comment:
`MetadataResult<Map<String, Set<MetadataKey>>>` may require a custom serializer, we can check this when implementing the agent service, but we should keep it in mind.